src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -12,23 +32,31 @@ export const TodoFilter = () => (
 
     <p className="control is-expanded has-icons-left has-icons-right">
       <input
+        value={titleFilter}
         data-cy="searchInput"
         type="text"
         className="input"
         placeholder="Search..."
+        onChange={(e) => handleTitleFilterChange(e.target.value)}
       />
       <span className="icon is-left">
         <i className="fas fa-magnifying-glass" />
       </span>
 
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button
-          data-cy="clearSearchButton"
-          type="button"
-          className="delete"
-        />
-      </span>
+      {
+        titleFilter
+        && (
+          <span className="icon is-right" style={{ pointerEvents: 'all' }}>
```

Comment:
Using inline styling without any specific reason is considered a bad practice. Let's move it away from here.

Reply:
I didn't change this line. It's a default string. I have an error without it.